### PR TITLE
Completed react-context exercise.-Christina Le

### PR DESCRIPTION
**Pull request by ChristinaL24**
Kapture 2022-04-07 at 10 29 37

Screen Shot 2022-04-07 at 11 34 28 AM

### CODE REVIEW

File: react-context/client/pages/home.jsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  gif: {
+    position: 'absolute'
+  }
+};
+
+export default class Home extends React.Component {
+  render() {
+
+    if (!this.context.user) return <Redirect to="sign-in" />;
+
```
Comment (thebearingedge) [inline react-context/client/pages/home.jsx:21]: This component needs a `contextType` as well.

File: react-context/flow.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+Create a file in the root of your react-context solution directory named flow.txt.
+For each feature in the application, describe the flow of the code in writing,
+including the client, server, and database. How does the code work?
+
+
+User can sign up.
+  -Client:
+    -The client will send a POST request to the server to '/api/auth/sign-up' and
```
Comment (thebearingedge) [inline react-context/flow.txt]: It's good that you split this up into Client server and database, but each feature in your write-up is only mostly covered. For many of the features, the interprocess communication goes

```
client -> req -> server -> query -> database -> result -> server -> res -> client
```

so try to fill out the later parts where the client receives the response from the server and then does something.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Good job so far, but we want to talk about what the client does when it receives a response from the client.

File: react-context/client/pages/home.jsx (hunk 1 above)
Comment (ChristinaL24) [reply react-context/client/pages/home.jsx:21]: rick

**Review by thebearingedge — APPROVED** [review]
Ok, very good. The "stay signed in" and "sign out" features are actually purely client-side code, since no HTTP request (beyond loading the HTML, CSS, and JS) is made when the user goes through them. Those features work entirely in the browser.

**Pull request merged**